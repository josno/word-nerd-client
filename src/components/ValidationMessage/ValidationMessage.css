.validation-message {
	font-size: 1rem;
	color: red;
	font-style: italic;
	margin: 0px;
}
