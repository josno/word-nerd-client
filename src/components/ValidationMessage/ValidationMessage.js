import React from 'react';
import './ValidationMessage.css';

export default function ValidationMessage(props) {
	return (
		<div className="validation-message-box">
			<p className="validation-message">{props.message}</p>
		</div>
	);
}
