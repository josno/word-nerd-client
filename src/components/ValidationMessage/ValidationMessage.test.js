import React from 'react';
import Enzyme from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ValidationMessage from './ValidationMessage';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('<Preview/>', () => {
	it('renders as expected', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<ValidationMessage />
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders nothing when no props given', () => {
		const wrapper = shallow(<ValidationMessage />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders with props', () => {
		const tree = renderer
			.create(<ValidationMessage message={'Validation Error String'} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
