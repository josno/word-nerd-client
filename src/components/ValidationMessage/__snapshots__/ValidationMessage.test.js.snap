// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preview/> renders as expected 1`] = `
<div
  className="validation-message-box"
>
  <p
    className="validation-message"
  />
</div>
`;

exports[`<Preview/> renders nothing when no props given 1`] = `
<div
  className="validation-message-box"
>
  <p
    className="validation-message"
  />
</div>
`;

exports[`<Preview/> renders with props 1`] = `
<div
  className="validation-message-box"
>
  <p
    className="validation-message"
  >
    Validation Error String
  </p>
</div>
`;
